src/session_base.cpp

```diff
@@ -409,6 +409,11 @@ void zmq::session_base_t::process_attach (i_engine *engine_)
         zmq_assert (!_pipe);
         _pipe = pipes[0];
 
+        //  The endpoints strings are not set on bind, set them here so that
+        //  events can use them.
+        pipes[0]->set_endpoint_pair (engine_->get_endpoint ());
+        pipes[1]->set_endpoint_pair (engine_->get_endpoint ());
```

Comment:
I am not sure, somehow it feels that the endpoint pair should be reversed on one of the pipes. But I don't oversee where this might have an effect.

Reply:
I guess it depends on how one interprets the local and remote attributes - if the reference point is the application thread then yes, but if the reference point is the socket itself then no

Reply:
Well yes, local and remote are the same wrt the socket. Still the direction is different. Hm. Let's see, when analysing the data from the monitoring :)